### PR TITLE
pbTests: Update VPC.sh to connect to 127.0.0.1:$port for Ansible && refactor

### PR DESCRIPTION
**Pull request by Willsparker**
Fixes: #1287 (@sxa  Would you prefer the issue is closed once the Jenkins issue has been fixed?)

The changes are as follows:
- `vagrantPlaybookCheck.sh` has been changed to connect to `127.0.0.1:$portNO` instead of the IP that Vagrant assigns to it. This is due to different users on the same machine assigning a VM with an IP that's already in use. This causes issues when trying to run the playbooks on them in parallel. Windows still uses the old IP method, as SSH port forwarding doesn't happen for Windows VMs. This is alright as this is the only OS that uses IP and so won't clash with any other one.
- `vagrantPlaybookCheck.sh` has also been slightly refactored to remove superfluous debug messages ( for instance, printing the time when the Playbook finishes, despite the `VPC` Jenkins job printing timestamps ), as well as make the code more readable. This includes updating old comments.
- `buildJDK.sh` has been updated to assign the `FILENAME` variable as this was stopping builds from starting when using `make-adopt-build-farm.sh`.
- Removing the code to generate `hosts.tmp` from each of the vagrantfiles, as those files aren't needed anymore.
- Removing the SUSE12 Vagrantfile as we no longer run it through `VPC`, and it's coming out of support fairly soon anyway.

PR marked as draft until I'm able to run this through `VPC`.

### CODE REVIEW

**Comment by Willsparker** [comment]
https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/724/
The PR seems to work! Ready for review :-)

File: ansible/pbTestScripts/vagrantPlaybookCheck.sh [hunk 1]
```diff
@@ -211,32 +212,27 @@ startVMPlaybook()
...
-	if [[ "$OS" == "FreeBSD12" || "$OS" == "Debian10" ]]; then
-               vagrant rsync-back
-	fi
-	# Generate hosts.unx file for Ansible to use, remove prior hosts.unx if there
-	[[ -f playbooks/AdoptOpenJDK_Unix_Playbook/hosts.unx ]] && rm playbooks/AdoptOpenJDK_Unix_Playbook/hosts.unx
-	cat playbooks/AdoptOpenJDK_Unix_Playbook/hosts.tmp | tr -d \\r | sort -nr | head -1 > playbooks/AdoptOpenJDK_Unix_Playbook/hosts.unx && rm playbooks/AdoptOpenJDK_Unix_Playbook/hosts.tmp
-	local vagrantIP=$(cat playbooks/AdoptOpenJDK_Unix_Playbook/hosts.unx)
+
+	rm -f playbooks/AdoptOpenJDK_Unix_Playbook/hosts.unx
+	echo "[127.0.0.1]:$(vagrant  port | grep host | awk '{ print $4 }')" >> playbooks/AdoptOpenJDK_Unix_Playbook/hosts.unx
```
Comment (karianna) [inline ansible/pbTestScripts/vagrantPlaybookCheck.sh:217]: double whitespace after vagrant

Comment (karianna) [reply ansible/pbTestScripts/vagrantPlaybookCheck.sh:217]: It's definitely $4 we want?

File: ansible/pbTestScripts/vagrantPlaybookCheck.sh [hunk 2]
```diff
@@ -211,32 +212,27 @@ startVMPlaybook()
...
 	cd $WORKSPACE/adoptopenjdkPBTests/$folderName-$branchName/ansible
+
 	if [[ "$testNativeBuild" = true && "$pb_failed" == 0 ]]; then
-		ssh -i $PWD/id_rsa vagrant@$vagrantIP "cd /vagrant/pbTestScripts && ./buildJDK.sh $buildURL $jdkToBuild $buildHotspot"
-		echo The build finished at : `date +%T`
+		ssh -p $(vagrant  port | grep host | awk '{ print $4 }') -i $PWD/id_rsa vagrant@127.0.0.1 "cd /vagrant/pbTestScripts && ./buildJDK.sh $buildURL $jdkToBuild $buildHotspot"
 		if [[ "$runTest" = true ]]; then
-	        	ssh -i $PWD/id_rsa vagrant@$vagrantIP "cd /vagrant/pbTestScripts && ./testJDK.sh"
-			echo The test finished at : `date +%T`
+			ssh -p $(vagrant  port | grep host | awk '{ print $4 }') -i $PWD/id_rsa vagrant@127.0.0.1 "cd /vagrant/pbTestScripts && ./testJDK.sh"
```
Comment (karianna) [inline ansible/pbTestScripts/vagrantPlaybookCheck.sh:235]: Could dedup some of this?

Comment (Willsparker) [reply ansible/pbTestScripts/vagrantPlaybookCheck.sh:235]: dedup?

File: ansible/pbTestScripts/vagrantPlaybookCheck.sh (hunk 1 above)
Comment (Willsparker) [reply ansible/pbTestScripts/vagrantPlaybookCheck.sh]: Yep, definitely! Tested here https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/724/ :-)

File: ansible/pbTestScripts/vagrantPlaybookCheck.sh (hunk 2 above)
Comment (karianna) [reply ansible/pbTestScripts/vagrantPlaybookCheck.sh]: Most of the ssh command was a repeat - but ignore my comment, deduping code in bash is actually fraught with danger.

**Comment by karianna** [comment]
@Willsparker - Travis failed though.

**Comment by sxa** [comment]
> @Willsparker - Travis failed though.

A quick look suggests that's a docker pull fail which isn't related to anything in this PR unless I'm mistaken @karianna ?

**Comment by sxa** [comment]
Is the `FILENAME` environment variable requirement new? I hit that in the last week too while doing something else.

**Comment by Willsparker** [comment]
> Is the FILENAME environment variable requirement new? I hit that in the last week too while doing something else.

Apparently so, though I'm not sure why. I added it into Windows 3 months ago ( https://github.com/AdoptOpenJDK/openjdk-infrastructure/pull/1263 ) , but I suppose it didn't affect Unix back then

**Pull request merged**